app/src/main/java/protect/card_locker/ScanActivity.java

```diff
@@ -36,6 +43,9 @@
  * originally licensed under Apache 2.0
  */
 public class ScanActivity extends CatimaAppCompatActivity {
+
+    private static int CAMERA_PERMISSION_REQUEST_CODE = 250;
```

Comment:
Could you explain where this 250 is coming from? I see it only used as result but never given as return code.

Reply:
> Could you explain where this 250 is coming from? I see it only used as result but never given as return code.

This is the request code that ` CaptureManager ` uses to apply for Camera Permission.

Reply:
I'd rather not depend on internal private variables, it would be better to use the `getCameraPermissionReqCode()` just to be sure that if they ever change it (unlikely, but possible) the code won't break:

https://github.com/journeyapps/zxing-android-embedded/blob/40260272fcff4f14181803495e7d370c23e35db7/zxing-android-embedded/src/com/journeyapps/barcodescanner/CaptureManager.java#L423-L425

I feel a bit weird about how both this class and the `CatimaCaptureManager` are listening to camera permission being allowed or denied.

---

_Wrong stuff here, but just to show my train of thought:_

Naively I would suggest to maybe add an extra argument to the `CatimaCaptureManager` constructor like:
```
public CatimaCaptureManager(Activity activity, DecoratedBarcodeView barcodeView, Callback onCameraFailure) {
    [...]
}
```

and have that callback show the UI for permission denied:

https://github.com/CatimaLoyalty/Android/blob/cae95d257768546dd4069c3a33f80d6112fa671d/app/src/main/java/protect/card_locker/CatimaCaptureManager.java#L19-L24

_End of wrong stuff_

---

However, the `CaptureManager` doesn't seem to notify on the permission suddenly being available so I suppose explicitly listening to the permission here is indeed the best choice and the only important part is using the `getCameraPermissionReqCode()` from the `CatimaCaptureManager` class instead of the hardcoded 250.

Reply:
It looks like it's the best practice to get the permission code from ` CaptureManager ` instead of hardcoding it, however, CaptureManager does not provide a function to get the member variable ` cameraPermissionReqCode ` and also since it's imported from the library ` com.journeyapps:zxing-android-embedded` , we may not be able to edit it.

Reply:
> however, CaptureManager does not provide a function to get the member variable cameraPermissionReqCode

Does it not? `getCameraPermissionReqCode()` seems public to me, so shouldn't `CatimaCaptureManager` have it given it extends `CaptureManager`?

Reply:
Oh sorry, I see it now. It's a static function.